MapleServer2/PacketHandlers/Game/SkillHandler.cs

```diff
@@ -0,0 +1,127 @@
+﻿using Maple2Storage.Types;
+using MaplePacketLib2.Tools;
+using MapleServer2.Constants;
+using MapleServer2.Packets;
+using MapleServer2.Servers.Game;
+using Microsoft.Extensions.Logging;
+
+namespace MapleServer2.PacketHandlers.Game
+{
+    public class SkillHandler : GamePacketHandler
+    {
+        public override RecvOp OpCode => RecvOp.SKILL;
+
+        public SkillHandler(ILogger<SkillHandler> logger) : base(logger) { }
+
+        private enum SkillHandlerMode : byte
+        {
+            FirstSent = 0x0,  // Start of a skill
+            Damage = 0x1,     // Damaging part of a skill. one is sent per hit
+            Mode3 = 0x3,
+            Mode4 = 0x4
+        }
+
+        private enum DamagingMode : byte
+        {
+            TypeOfDamage = 0x0,
+            AoeDamage = 0x1,
+            TypeOfDamage2 = 0x2
+        }
+
+        public override void Handle(GameSession session, PacketReader packet)
+        {
+            short skillLevel;
+            long skillCount;
+            int skillId;
+            int value;
+            int playerObjectId;
+            CoordF coords;
+            CoordF coords2;
+            CoordF coords3;
+
+            SkillHandlerMode mode = (SkillHandlerMode)packet.ReadByte();
+            switch (mode)
+            {
+                case SkillHandlerMode.FirstSent:
```

Comment:
Can each mode be handled in a different method?

Reply:
Sure!